usaspending_api/awards/v2/views/idvs/awards.py

```diff
@@ -0,0 +1,151 @@
+        tf.ordering_period_end_date                last_date_to_order,
+        ac.total_obligation                        obligated_amount,
+        ac.period_of_performance_current_end_date,
+        ac.period_of_performance_start_date,
+        ac.piid
+    from
+        parent_award pap
+        inner join awards ap on ap.id = pap.award_id
+        inner join awards ac on ac.fpds_parent_agency_id = ap.fpds_agency_id and ac.parent_award_piid = ap.piid and
+            ac.type not like 'IDV\_%%'
+        inner join transaction_fpds tf on tf.transaction_id = ac.latest_transaction_id
+    where
+        pap.{award_id_column} = %s
+    order by
+        {sort_column} {sort_direction}, ac.id {sort_direction}
+    limit {limit} offset {offset}
+""")
+
+
+def _prepare_tiny_shield_rules():
+    """
+    Our TinyShield rules never change.  Encapsulate them here and store them
+    once in TINY_SHIELD_RULES.
+    """
+
+    # This endpoint supports paging.
+    models = deepcopy(PAGINATION)
+
+    # Add the list of sortable columns for validation.
+    sort_rule = TinyShield.get_model_by_name(models, 'sort')
+    sort_rule['type'] = 'enum'
+    sort_rule['enum_values'] = SORTABLE_COLUMNS
+    sort_rule['default'] = DEFAULT_SORT_COLUMN
+
+    # Add additional models for the award id and the idv filter.
+    models.extend([
+        # Award id can be either an integer or a string depending upon whether
+        # it's an internal id or a unique generated id.
+        {'key': 'award_id', 'name': 'award_id', 'type': 'any', 'optional': False, 'models': [
+            {'type': 'integer'},
+            {'type': 'text', 'text_type': 'search'}
+        ]},
+        {'key': 'idv', 'name': 'idv', 'type': 'boolean', 'default': True, 'optional': True}
+    ])
+
+    return models
+
+
+TINY_SHIELD_RULES = _prepare_tiny_shield_rules()
+
+
+class IDVAwardsViewSet(APIDocumentationView):
+    """Returns the direct children of an IDV.
+    endpoint_doc: /awards/idvs/awards.md
+    """
+
+    @staticmethod
+    def _parse_and_validate_request(request: Request) -> dict:
+        return TinyShield(deepcopy(TINY_SHIELD_RULES)).block(request)
+
+    @staticmethod
+    def _business_logic(request_data: dict) -> list:
+        award_id, award_id_type = detect_award_id_type(request_data['award_id'])
+        award_id_column = 'award_id' if award_id_type is AwardIdType.internal else 'generated_unique_award_id'
+        sql = GET_IDVS_SQL if request_data['idv'] else GET_CONTRACTS_SQL
+        sql = sql.format(
+            award_id_column=Identifier(award_id_column),
+            award_id=Literal(award_id),
+            sort_column=Identifier(request_data['sort']),
+            sort_direction=SQL(request_data['order']),
+            limit=Literal(request_data['limit'] + 1),
+            offset=Literal((request_data['page'] - 1) * request_data['limit']),
+        )
+        with connection.cursor() as cursor:
+            cursor.execute(sql, [award_id])
```

Comment:
I encountered a 500 with this request (AttributeError: 'Composed' object has no attribute 'lower')

**URL**: `/api/v2/awards/idvs/awards/`
**BODY**: `{"award_id":"CONT_AW_9700_9700_71T0_SPM30008D3155"}`

Edit:
This was only when DEBUG=True in settings.py. Confusing why it occurs, should be investigated

Reply:
Man.  Good catch.  This is a problem with `django-debug-toolbar`.  It is trying to show off the SQL which it has incorrectly assumed is a string instead of a `psycopg2` SQL object.  I don't usually run with `django-debug-toolbar` enabled (mainly because it's slow - but now I have another reason!).

We can simply materialize the SQL before passing it into Django.  Bingo bango, problem solved!